### ISSUE TITLE
Document how to set up a non-cluster-admin role & rolebinding for Ark users

### ISSUE THREAD

**Issue by ncdc** [heptio/ark#350]
Consider `kubectl create role/rolebinding` and/or raw yaml

```
kubectl -n heptio-ark create role ark --verb='*' --resource='backups.ark.heptio.com,restores.ark.heptio.com,schedules.ark.heptio.com,config.ark.heptio.com'

kubectl -n heptio-ark create rolebinding --role=ark --user=YOUR_USER_HERE
```

```yaml
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: Role
metadata:
  namespace: heptio-ark
  name: ark
  labels:
    component: ark
rules:
  - apiGroups:
      - ark.heptio.com
    verbs:
      - "*"
    resources:
      - "*"

---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: RoleBinding
metadata:
  name: ark
subjects:
  - kind: User
    name: YOUR_USER_HERE
roleRef:
  kind: Role
  name: ark
  apiGroup: rbac.authorization.k8s.io
```

**Comment by gleeco** [comment]
wondering if this issue is still relevant?

**Comment by ncdc** [comment]
@gleeco it depends on whether or not you want to run Ark as cluster-admin or as a lesser-privileged user/serviceaccount. We recommend running as cluster-admin as that allows Ark to be able to back up and restore anything in the cluster, from any namespace. But we also recognize that that gives Ark broad powers, and you may decide it's too broad. That's what this issue is for.

**Comment by Bradamant3** [comment]
This seems timely -- we've been documenting elsewhere the advantages of namespace-specific backups/restores, and this should be a part of that story.

**Comment by Bradamant3** [comment]
@ncdc question: Does Ark need all verbs? Access to all resources? Would it be useful to provide a more fine-grained example, or is such not relevant? IOW, can you effectively limit only per namespace, not per actions on resources? (Sigh must recollect/relearn more about Ark ...)

**Comment by ncdc** [comment]
@Bradamant3 we'll need to review the backup and restore code to provide a detailed breakdown of which verbs are used, and when.

For rbac in general, you can limit by namespace and action and resource (e.g. in namespace foo, I can create widgets).

If we do document a non cluster-admin configuration, where Ark only has access to a single namespace, **note** Ark will be unable to snapshot/restore PVs (these are cluster-scoped).

**Comment by Bradamant3** [comment]
We can keep the docs general about verbs and resources for now, too. But yeah, I'm aware of the general rbac rules pattern, that's why I asked. Thanks for the reminder about PV limitations.

**Comment by skriss** [comment]
Closed by #1023
